www/js/survey/multilabel/MultiLabelButtonGroup.tsx

```diff
@@ -151,16 +191,47 @@ const MultilabelButtonGroup = ({ trip, buttonsInline = false }) => {
             </Dialog.Title>
             <Dialog.Content style={{ maxHeight: windowHeight / 2, paddingBottom: 0 }}>
               <ScrollView style={{ paddingBottom: 24 }}>
+                <Text style={{ fontSize: 12, color: '#333333', paddingVertical: 4 }}>
+                  Default Mode
+                </Text>
                 <RadioButton.Group onValueChange={(val) => onChooseLabel(val)} value={chosenLabel}>
-                  {labelOptions?.[modalVisibleFor]?.map((o, i) => (
+                  {labelOptions?.[modalVisibleFor]?.map((o, i) => {
+                    // This code is designed to order the modes as follows: default modes -> customized modes -> other
+                    if (o.value === 'other') return;
+                    return (
+                      // @ts-ignore
+                      <RadioButton.Item
+                        key={i}
+                        label={t(o.text)}
+                        value={o.value}
+                        style={{ paddingVertical: 2 }}
+                      />
+                    );
+                  })}
+                  {customModes.length > 0 && (
+                    <>
+                      <Divider style={{ marginVertical: 10 }} />
+                      <Text style={{ fontSize: 12, color: '#333333', paddingVertical: 4 }}>
+                        Custom Mode
+                      </Text>
+                    </>
+                  )}
+                  {customModes.map((key, i) => (
                     // @ts-ignore
                     <RadioButton.Item
                       key={i}
-                      label={t(o.text)}
-                      value={o.value}
+                      label={getProcessedText(key)}
+                      value={key}
                       style={{ paddingVertical: 2 }}
                     />
                   ))}
+                  <Divider style={{ marginVertical: 10 }} />
+                  <RadioButton.Item
+                    key="Other"
+                    label={t('Other')}
+                    value="other"
+                    style={{ paddingTop: 10 }}
+                  />
                 </RadioButton.Group>
```

Comment:
Instead of skipping 'other' and tacking it back on at the end, here is what I would recommend.
Go through the list as normal and when you encounter 'other', just prepend it with the whole 'custom modes' section

```tsx
  <RadioButton.Group onValueChange={(val) => onChooseLabel(val)} value={chosenLabel}>
    {labelOptions?.[modalVisibleFor]?.map((o, i) => {
      const radioItemForOption = (
        <RadioButton.Item
          key={i}
          label={t(o.text)}
          value={o.value}
          style={{ paddingVertical: 2 }}
        />
      );
      /* if this is the 'other' option and there are some custom modes,
        show the custom modes section before 'other' */
      if (o.value == 'other' && customModes.length) {
        return (
          <>
            <Divider style={{ marginVertical: 10 }} />
            <Text style={{ fontSize: 12, color: '#333333', paddingVertical: 4 }}>
              Custom Mode
            </Text>
            {customModes.map((key, i) => (
              <RadioButton.Item
                key={i}
                label={getProcessedText(key)}
                value={key}
                style={{ paddingVertical: 2 }}
              />
            ))}
            <Divider style={{ marginVertical: 10 }} />
            {radioItemForOption}
          </>
        );
      }
      // otherwise, just show the radio item as normal
      return radioItemForOption;
    })}
  </RadioButton.Group>
```

Reply:
This will also work in the event that any future program opts not to have 'other' as an option

Reply:
@JGreenlee Thank you! I love this code! Brilliant 🥇